crates/compiler/parse/src/ast.rs

```diff
@@ -2326,15 +2342,31 @@ macro_rules! impl_extract_spaces {
 
         impl<'a, $($($generic_args: Copy),*)?> ExtractSpaces<'a> for $t<'a, $($($generic_args),*)?> {
             type Item = Self;
-            fn extract_spaces(&self) -> Spaces<'a, Self::Item> {
+            fn extract_spaces(&self, arena: &'a Bump) -> Spaces<'a, Self::Item> {
                 match self {
                     $t::SpaceBefore(item, before) => {
                         match item {
```

Comment:
Rather than doing a strict 2-level match here, we should change to doing a recursive call, similar to what happens in `expr_lift_spaces` (and similar functions) from the formatter.

Reply:
Is this not recursive?  We call extract spaces on the item on line 2354